massa-signature/src/signature_impl.rs

```diff
@@ -899,13 +902,17 @@ pub fn verify_signature(
 }
 
 /// Generate a random private key from a RNG.
-pub fn generate_random_private_key() -> PrivateKey {
+pub fn generate_random_keypair() -> KeyPair {
     use secp256k1::rand::rngs::OsRng;
     let mut rng = OsRng::new().expect("OsRng");
-    PrivateKey(secp256k1::KeyPair::from_secret_key(
+    let keypair = secp256k1::KeyPair::from_secret_key(
```

Comment:
the lib has a native keypair generator: https://docs.rs/secp256k1/latest/secp256k1/struct.KeyPair.html#examples

Reply:
generally, we shouldn't be using key derivation anymore

Reply:
I changed to use it.